static/app/utils/useUndoableReducer.tsx

```diff
@@ -72,12 +72,27 @@ export function useUndoableReducer<
 >(
   reducer: R,
   initialState: ReducerState<R>
-): [ReducerState<R>, React.Dispatch<UndoableReducerAction<ReducerAction<R>>>] {
+): [
+  ReducerState<R>,
+  React.Dispatch<UndoableReducerAction<ReducerAction<R>>>,
+  {
+    peekFuture: () => ReducerState<R> | undefined;
+    peekHistory: () => ReducerState<R> | undefined;
+  }
+] {
   const [state, dispatch] = useReducer(makeUndoableReducer(reducer), {
     current: initialState,
     previous: undefined,
     next: undefined,
   });
 
-  return [state.current, dispatch];
+  const peekHistory = useCallback((): ReducerState<R> | undefined => {
+    return state?.previous?.current;
+  }, [state.current]);
+
+  const peekFuture = useCallback((): ReducerState<R> | undefined => {
+    return state?.next?.current;
+  }, [state.current]);
+
+  return [state.current, dispatch, {peekHistory, peekFuture}];
```

Comment:
Oh yeah ofc. I think I did it this way because I wanted to maybe extend this to a more generator like approach where you could iterate over all previous/next states

Reply:
Altho since it's a linked list anyways, I guess you could already do that